engines/wasmer/wasmer.go

```diff
@@ -95,28 +97,40 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
-
-func Engine() wapc.Engine {
-	return &engineInstance
+// WithEngine allows you to override the default engine. Defaults to
+// `wasmer.NewEngine`.
+func WithEngine(engine *wasmer.Engine) *wasmer.Engine {
+	return engine
 }
 
+func Engine(engs ...*wasmer.Engine) wapc.Engine {
```

Comment:
this feels a bit odd, is this common (using varargs to make one of the args optional)?

I think how wazero does it is a lot less confusing (having a second function vs being too clever with one)

```
func EngineWithRuntime(newRuntime NewRuntime) wapc.Engine {
```

Reply:
@codefromthecrypt I was looking at that too but, if I remember, there was a comment regarding removing the function and just passing the engine ptr.  Without using the varargs syntax you would have to pass nil if you want a new engine.  I thought that was hideous looking but I am certainly willing to make it a function as orig, I just don't want to reignite the comment of removing the function for the ptr. I can make it look like wazero if that is desirable?

Reply:
Works for me!  I'll give it a whirl.

Reply:
@codefromthecrypt.  Changes pushed